homeassistant/components/open_meteo/weather.py

```diff
@@ -25,7 +32,12 @@ async def async_setup_entry(
 ) -> None:
     """Set up Open-Meteo weather entity based on a config entry."""
     coordinator = hass.data[DOMAIN][entry.entry_id]
-    async_add_entities([OpenMeteoWeatherEntity(entry=entry, coordinator=coordinator)])
+    async_add_entities(
+        [
+            OpenMeteoWeatherEntity(entry=entry, coordinator=coordinator, hourly=False),
+            OpenMeteoWeatherEntity(entry=entry, coordinator=coordinator, hourly=True),
```

Comment:
I don't think adding two `weather` entities is a good method. It only adds confusion.

Reply:
How would you do it? I also think that ideally this should be one entity. But his method does allow you to easily display the data in frontend cars that work with an hourly forecast weather entity such as the default one in HA. I could only find  the [met.no integration](https://www.home-assistant.io/integrations/met/) that also has both daily and hourly forecasts and it solved it in this way.

Reply:
This PR <https://github.com/home-assistant/core/pull/75219> allows weather entities to provide hourly, bi-daily and daily forecasts. Instead of adding a new entity, I'd suggest this PR is refactored once that PR is merged.

Reply:
> This PR #75219 allows weather entities to provide hourly, bi-daily and daily forecasts. Instead of adding a new entity, I'd suggest this PR is refactored once that PR is merged.

Agreed, I already asked about the state of that PR [here](https://github.com/home-assistant/core/pull/75219#issuecomment-1561052680). It's been in process for a year now so I wonder when it will be merged.

Reply:
Great. Can you give a preview of what it will look like in the frontend?

Reply:
There are some screenshots in the frontend PR: https://github.com/home-assistant/frontend/pull/15028

Reply:
Looks good. As we're reworking the forecasting system maybe it's also a good idea to consider my request regarding cloud coverage property [here](https://github.com/home-assistant/architecture/discussions/905).